roachtest: assert against bogus return from pgurls

tbg opened the pull request:
See https://github.com/cockroachdb/cockroach/issues/48091.

Apparently `c.InternalAddr` can return an empty addr:

https://github.com/cockroachdb/cockroach/blob/26b9716f3d1b30cb8456fc8f1de1bad78c4d26fa/pkg/cmd/roachtest/cluster.go#L2112-L2134

I have no idea how this can happen despite a bit of sleuthing, but
apparently it does, so the low cost way to home in on the problem
is to sprinkle some assertions.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48156)
<!-- Reviewable:end -->

andreimatei reviewed (commented):
Isn't the assertion in `roachprod` enough?

LGTM, but I hope we don't need 3 of them :)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48156#-:-M67489r22pj5OJt-wOj:bh22frk)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @jlinder)



<!-- Sent from Reviewable.io -->

tbg committed:
roachtest: assert against bogus return from pgurls

See https://github.com/cockroachdb/cockroach/issues/48091.

I have no idea how this can happen despite a bit of sleuthing, but
apparently it does, so the low cost way to home in on the problem
is to sprinkle some assertions.

Release note: None

tbg commented:
TFTR! I'll take my chances with the extra assertions. It's just too unclear where things go wrong :-/ though I also do expect the roachprod one to fire first.

bors r=andreimatei
